waffle_utils/dataset/dataset.py

```diff
@@ -199,40 +204,35 @@ def from_coco(
 
         # parse coco annotation file
         coco = io.load_json(coco_file)
-        ds.add_imgs([Image.from_dict(image) for image in coco["images"]])
-        ds.add_anns(
-            [
-                Annotation.from_dict(annotation)
-                for annotation in coco["annotations"]
-            ]
-        )
-        ds.add_cats(
-            [Category.from_dict(category) for category in coco["categories"]]
-        )
+        for image_dict in coco["images"]:
+            image_id = image_dict.pop("id")
```

Comment:
Is there a philosophy of using `dict.pop("key")` instead of `dict["key"]` or `dict.get("key")`? If this dictionary placeholder needs to be reused, it will ruin that placeholder. Are you using the `pop()` on purpose?

Reply:
This is for remove `id` from `image_dict`.
`from_dict` does not recieve `id` because `id` is python built-in function.